### PR TITLE
refactor: Centralise BigNumber import

### PR DESCRIPTION
**Pull request by pxrl**
Stop depending directly on ethers for this, such that it should be easier to eventually shift away from ethers v5. This has been repeated upstream in the SDK here: https://github.com/across-protocol/sdk/pull/725

### CODE REVIEW

File: src/adapter/bridges/ZKSyncBridge.ts [hunk 1]
```diff
@@ -176,7 +176,7 @@ export class ZKSyncBridge extends BaseBridgeAdapter {
     return l2Gas;
   }
 
-  protected getMailboxContract() {
+  protected getMailboxContract(): Contract {
```
Comment (pxrl) [inline src/adapter/bridges/ZKSyncBridge.ts:179]: This was just a driveby fix that popped up when looking at the eslint output.

**Comment by pxrl** [comment]
Pulling back to draft because this completely explodes in CI.

File: .eslintrc.js [hunk 2]
```diff
@@ -38,6 +38,13 @@ module.exports = {
     "no-duplicate-imports": "error",
     "@typescript-eslint/no-floating-promises": ["error"],
     "@typescript-eslint/no-misused-promises": ["error", { checksVoidReturn: false }],
+    "no-restricted-imports": [
+      "error",
+      {
+        patterns: [{ group: ["@ethersproject/bignumber"], message: "Use 'src/utils/SDKUtils' instead" }],
+        paths: [{ name: "ethers", importNames: ["BigNumber"], message: "Use 'src/utils/SDKUtils' instead" }],
```
Comment (pxrl) [inline .eslintrc.js:45]:
```suggestion
        patterns: [{ group: ["@ethersproject/bignumber"], message: "Use 'src/utils/BNUtils' instead" }],
        paths: [{ name: "ethers", importNames: ["BigNumber"], message: "Use 'src/utils/BNUtils' instead" }],
```

File: src/interfaces/Token.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
-import { BigNumber } from "ethers";
 import { SortableEvent } from ".";
+import { BigNumber } from "../utils";
```
Comment (pxrl) [inline src/interfaces/Token.ts:2]:
```suggestion
import { BigNumber } from "../utils";
import { SortableEvent } from ".";
```

File: src/monitor/Monitor.ts [hunk 4]
```diff
@@ -42,11 +43,9 @@ import { MonitorClients, updateMonitorClients } from "./MonitorClientHelper";
 import { MonitorConfig } from "./MonitorConfig";
 import { CombinedRefunds } from "../dataworker/DataworkerUtils";
 
-export const REBALANCE_FINALIZE_GRACE_PERIOD = process.env.REBALANCE_FINALIZE_GRACE_PERIOD
-  ? Number(process.env.REBALANCE_FINALIZE_GRACE_PERIOD)
-  : 60 * 60;
 // 60 minutes, which is the length of the challenge window, so if a rebalance takes longer than this to finalize,
 // then its finalizing after the subsequent challenge period has started, which is sub-optimal.
+export const REBALANCE_FINALIZE_GRACE_PERIOD = Number(process.env.REBALANCE_FINALIZE_GRACE_PERIOD ?? 60 * 60);
```
Comment (pxrl) [inline src/monitor/Monitor.ts:48]: This is just a driveby change.

File: .eslintrc.js [hunk 5]
```diff
@@ -38,6 +38,13 @@ module.exports = {
     "no-duplicate-imports": "error",
     "@typescript-eslint/no-floating-promises": ["error"],
     "@typescript-eslint/no-misused-promises": ["error", { checksVoidReturn: false }],
+    "no-restricted-imports": [
```
Comment (james-a-morris) [inline .eslintrc.js:41]: +1 I like the notion of using the linter to handle this type of use case

**Review by james-a-morris — COMMENTED** [review]
Is the goal to move all ethers imports or just BigNumber?

**Comment by pxrl** [comment]
> Is the goal to move all ethers imports or just BigNumber?

It's just BigNumber for now; this is one of the major hurdles for moving away from ethers v5. There could be other cases though, so this is a pattern I might repeat if I find another blocker.

File: src/utils/SDKUtils.ts [hunk 6]
```diff
@@ -2,7 +2,6 @@ import * as sdk from "@across-protocol/sdk";
 
 export class BlockFinder extends sdk.utils.BlockFinder {}
 export type BlockFinderHints = sdk.utils.BlockFinderHints;
-
```
Comment (james-a-morris) [inline src/utils/SDKUtils.ts:5]: Do we need this?

**Review by james-a-morris — APPROVED** [review]
One nit

File: src/utils/SDKUtils.ts (hunk 6 above)
Comment (pxrl) [reply src/utils/SDKUtils.ts]: Dropped here: https://github.com/across-protocol/relayer/pull/1820/commits/221ae769d2bcef55dc4c2ac6da93a99ae031c824
